pycode/memilio-epidata/memilio/epidata/getPopulationData.py

```diff
@@ -229,142 +228,317 @@ def get_population_data(read_data=dd.defaultDict['read_data'],
+    This data can either be downloaded automatically or read from
+    "out_folder"/Germany/ if it was downloaded before.
+
+    @param read_data False or True. Defines if data is read from file or
+        downloaded. Default defined in defaultDict.
+    @param file_format File format which is used for writing the data.
+        Default defined in defaultDict.
+    @param out_folder Path to folder where data is written in folder
+        out_folder/Germany. Default defined in defaultDict.
+    @param no_raw True or False. Defines if unchanged raw data is written or
+        not. Default defined in defaultDict.
+    @param split_gender [Default: False] or True. Defines whether data is
+        splitted by gender
+    @param merge_eisenach [Default: True] or False. Defines whether the
+        counties 'Wartburgkreis' and 'Eisenach' are listed separately or
+        combined as one entity 'Wartburgkreis'.
+    @return DataFrame with adjusted population data for all ages to current level.
     """
-    counties, zensus, reg_key = load_population_data(
-        out_folder, read_data=read_data, no_raw=no_raw,
-        file_format=file_format)
-
-    # find region keys for census population data
-    key = np.zeros((len(zensus)))
-    for i in range(len(key)):
-        for j in range(len(reg_key)):
-            if zensus['Name'].values[i] == reg_key['NAME'].values.astype(str)[
-                    j]:
-                if zensus[dd.invert_dict(dd.GerEng)[dd.EngEng['population']]].values[i] == round(
-                        reg_key['Zensus_EWZ'].values[j] * 1000):
-                    key[i] = reg_key['AGS'].values[j]
-
-    inds = np.unique(key, return_index=True)[1]
-    # columns of downloaded data which should be replaced
-    male = ['M_Unter_3', 'M_3_bis_5', 'M_6_bis_14', 'M_15_bis_17',
-            'M_18_bis_24', 'M_25_bis_29', 'M_30_bis_39', 'M_40_bis_49',
-            'M_50_bis_64', 'M_65_bis_74', 'M_75_und_aelter']
-    female = ['W_Unter_3', 'W_3_bis_5', 'W_6_bis_14', 'W_15_bis_17',
-              'W_18_bis_24', 'W_25_bis_29', 'W_30_bis_39', 'W_40_bis_49',
-              'W_50_bis_64', 'W_65_bis_74', 'W_75_und_aelter']
-
-    if not split_gender:
-        # get data from zensus file and add male and female population data
-        data = np.zeros((len(inds), len(male)+2))
-        data[:, 0] = key[inds].astype(int)
-        data[:, 1] = zensus[dd.invert_dict(
-            dd.GerEng)[dd.EngEng['population']]].values[inds].astype(int)
-        for i in range(len(male)):
-            data[:, i+2] = zensus[male[i]].values[inds].astype(
-                int) + zensus[female[i]].values[inds].astype(int)
-
-        # define new columns for dataframe
-        columns = [
+    directory = dd.defaultDict['out_folder'].split('/pydata')[0]
+    filename = '12411-02-03-4'  # '12411-09-01-4-B'
+    new_data_file = os.path.join(directory, filename)
+    new_data_avail = os.path.isfile(new_data_file + '.xlsx')
+
+    if new_data_avail:
```

Comment:
Basically you are right. However, so far we only open an xlsx here while in the other 'load_population' a lot of jobs are done. The major problem here is rather that this had to be redesigned globally. We should discuss how and when this new data is loaded, how we actually call the different population files etc. This could be discussed in a future sprint meeting.